src/main/kotlin/com/manimdsl/errorhandling/semanticerror/SemanticErrorHandling.kt

```diff
@@ -1,3 +1,45 @@
 package com.manimdsl.errorhandling.semanticerror
 
-// TODO Add error handling functions for semantic analysis
\ No newline at end of file
+import com.manimdsl.errorhandling.ErrorHandler.addSemanticError
+import com.manimdsl.frontend.Type
+import org.antlr.v4.runtime.ParserRuleContext
+
+fun declareAssignError(action: String, rhsType: Type, lhsType: Type,
+                       ctx: ParserRuleContext) {
+    addSemanticError("$action $rhsType to $lhsType", getErrorLinePos(ctx))
+}
+
+fun redeclarationError(variable: String, variableType: Type,
+                       ctx: ParserRuleContext) {
+    addSemanticError("$variable of type $variableType is already declared", getErrorLinePos(ctx))
+}
+
+fun undeclaredAssignError(variable: String,
+                       ctx: ParserRuleContext) {
+    addSemanticError("$variable has not been declared", getErrorLinePos(ctx))
+}
+
+fun nonDataStructureMethodError(identifier: String, ctx: ParserRuleContext){
+    addSemanticError("$identifier is not a Data Structure", getErrorLinePos(ctx))
+}
+
+fun unsupportedMethodError(dataStructureType: String, method: String, ctx: ParserRuleContext){
+    addSemanticError("$dataStructureType does not support $method method", getErrorLinePos(ctx))
+}
+
+fun numOfArgsInMethodCallError(dataStructureType: String, method: String, numArgs: Int, ctx: ParserRuleContext){
+    // To modify once nex version is merged
+    addSemanticError("$method method on $dataStructureType does not accept $numArgs arguments", getErrorLinePos(ctx))
+}
+
+fun typeOfArgsInMethodCallError(dataStructureType: String, method: String, argType: String, argName:String, ctx: ParserRuleContext){
+    // To modify once nex version is merged
+    addSemanticError("$method method on $dataStructureType does not accept argument $argName of type $argType", getErrorLinePos(ctx))
+}
+
+/* Helper function that returns line and character position for errors */
+private fun getErrorLinePos(ctx: ParserRuleContext): String {
+    val line = ctx.getStart().line
```

Comment:
Is this not duplicated in syntax error handler

Reply:
As in the line number isn't in the syntax error listener?